specs/Standards/Wallets/InjectedWallets.md

```diff
@@ -0,0 +1,356 @@
+**NOTE: Do not replace the entire `window.near` object with your wallet implementation, or add any objects as properties of the `window.near` object that do not conform to the Injected Wallet Standard**
+
+At the core of a wallet are [`signTransaction`](#signtransaction) and [`signTransactions`](#signtransactions). These methods, when given a [`Transaction`](https://nomicon.io/RuntimeSpec/Transactions) instance from [`near-api-js`](https://github.com/near/near-api-js), will prompt the user to sign with a key pair previously imported (with the assumption it has [`FullAccess`](https://nomicon.io/DataStructures/AccessKey) permission).
+
+In most cases, a dApp will need a reference to an account and associated public key to construct a [`Transaction`](https://nomicon.io/RuntimeSpec/Transactions). The [`connect`](#connect) method helps solve this issue by prompting the user to select one or more accounts they would like to make visible to the dApp. When at least one account is visible, the wallet considers the dApp [`connected`](#connected) and they can access a list of [`accounts`](#accounts) containing an `accountId` and `publicKey`.
+
+For dApps that often sign gas-only transactions, [`FunctionCall`](https://nomicon.io/DataStructures/AccessKey#accesskeypermissionfunctioncall) access keys can be added/deleted for one or more accounts using the [`signIn`](#signin) and [`signOut`](#signout) methods. While this functionality could be achieved with [`signTransactions`](#signtransactions), it suggests a direct intention that a user wishes to sign in/out of a dApp's smart contract.
+
+### Wallet API
+
+Below is the entire API for injected wallets. It makes use of `near-api-js` to enable interoperability with dApps that will already use it for constructing transactions and communicating with RPC endpoints.
+
+```ts
+import { transactions, utils } from "near-api-js";
+
+interface Account {
+  accountId: string;
+  publicKey: utils.PublicKey;
+}
+
+interface Network {
+  networkId: string;
+  nodeUrl: string;
+}
+
+interface SignInParams {
+  permission: transactions.FunctionCallPermission;
+  accounts: Array<Account>;
+}
+
+interface SignOutParams {
+  accounts: Array<Account>;
+}
+
+interface TransactionOptions {
+    receiverId: string;
+    actions: Array<transactions.Action>;
+    signerId?: string;
+}
+
+interface SignTransactionParams {
+  transaction: TransactionOptions;
+}
+
+interface SignTransactionsParams {
+  transactions: Array<TransactionOptions>;
+}
+
+interface Events {
+  accountsChanged: { accounts: Array<Account> };
+}
+
+interface ConnectParams {
+    networkId: string;
+}
+
+type Unsubscribe = () => void;
+
+interface Wallet {
```

Comment:
As @PinkiNice and I mentioned in the previous NEP version (https://github.com/near/NEPs/pull/370#discussion_r987196550):

I think it could be good to also allow wallets to list their supported Networks. As this would allow dApps to avoid trying to `connect()` to wallets that don't support their app network.

It could be good to include a property in the Wallet interface such as:

```ts
interface Wallet {
  supportedNetworkIds: string[];
}
```

Or maybe something easier for the dApp to consume, like a function they can call:

```ts
type SupportsNetwork = (networkId: string) => boolean;

interface Wallet {
  // ...
  supportsNetwork: SupportsNetwork ;
}
```

Reply:
:+1: Thanks for the heads up -- I've added the `supportsNetwork` function to the standard; I like the idea of a function in this case rather than a static array as it feels more flexible and the dApp is probably going to know what networkId they are checking for support of.

I defined it as an async method that resolved to a boolean rather than the sync method you had sketched out, for flexibility on the wallet side for how it determines if a passed networkId is supported.  This should insulate us from needing to make a breaking API change or fragment the API if a wallet implementor needs to do something async 'behind the scenes' in order to to determine the result of this call.